### PR TITLE
Fixes related to loading of env variables

### PR DESCRIPTION
**Pull request by DonJayamanne**
Part of #10359

### CODE REVIEW

File: src/kernels/jupyter/jupyterKernelService.node.ts [hunk 1]
```diff
@@ -276,75 +271,38 @@ export class JupyterKernelService implements IJupyterKernelService {
...
-                // Ensure the python env folder is always at the top of the PATH, this way all executables from that env are used.
-                // This way shell commands such as `!pip`, `!python` end up pointing to the right executables.
-                // Also applies to `!java` where java could be an executable in the conda bin directory.
-                if (specModel.env) {
-                    this.envVarsService.prependPath(specModel.env as {}, path.dirname(interpreter.uri.fsPath));
-                }
-
-                // If user asks us to, set PYTHONNOUSERSITE
-                // For more details see here https://github.com/microsoft/vscode-jupyter/issues/8553#issuecomment-997144591
-                // https://docs.python.org/3/library/site.html#site.ENABLE_USER_SITE
```
Comment (DonJayamanne) [inline src/kernels/jupyter/jupyterKernelService.node.ts:312]: This duplicate code can be replaced with `KernelEnvironmentVariablesService.getEnvironmentVariables`

File: src/kernels/raw/launcher/kernelEnvVarsService.node.ts [hunk 2]
```diff
@@ -37,13 +44,7 @@ export class KernelEnvironmentVariablesService {
         let kernelEnv = kernelSpec.env && Object.keys(kernelSpec.env).length > 0 ? kernelSpec.env : undefined;
 
         // If an interpreter was not explicitly passed in, check for an interpreter path in the kernelspec to use
-        if (!interpreter) {
-            if (!kernelSpec.interpreterPath) {
-                traceInfo(
-                    `No custom variables for Kernel as interpreter path is not defined for kernel ${kernelSpec.display_name}`
-                );
-                return kernelEnv;
```
Comment (DonJayamanne) [inline src/kernels/raw/launcher/kernelEnvVarsService.node.ts:45]: We will always return the env variables in this method, never `undefined`
i.e. merge process.env with kernelspec.env and the like.

File: src/kernels/raw/launcher/kernelEnvVarsService.node.ts [hunk 3]
```diff
@@ -63,26 +64,17 @@ export class KernelEnvironmentVariablesService {
...
         ]);
-        if (!interpreterEnv && Object.keys(customEditVars || {}).length === 0) {
+        if (!interpreterEnv && Object.keys(customEnvVars || {}).length === 0) {
             traceInfo('No custom variables nor do we have a conda environment');
-            // Ensure the python env folder is always at the top of the PATH, this way all executables from that env are used.
-            // This way shell commands such as `!pip`, `!python` end up pointing to the right executables.
-            // Also applies to `!java` where java could be an executable in the conda bin directory.
-            if (interpreter) {
-                const env = kernelEnv || process.env;
-                this.envVarsService.prependPath(env, path.dirname(interpreter.uri.fsPath));
```
Comment (DonJayamanne) [inline src/kernels/raw/launcher/kernelEnvVarsService.node.ts:73]: This isn't necessary as we already do this further below (exact same code)

File: src/kernels/jupyter/jupyterKernelService.node.ts (hunk 1 above)
Comment (DonJayamanne) [reply src/kernels/jupyter/jupyterKernelService.node.ts:312]: Now both raw kernels and jupyter kernels will use the same set of code to generate env variables for staring of kernels.

File: src/kernels/raw/launcher/kernelEnvVarsService.node.ts [hunk 4]
```diff
@@ -96,16 +88,16 @@ export class KernelEnvironmentVariablesService {
...
-            this.envVarsService.appendPath(mergedVars, customEditVars[otherEnvPathKey]!);
+        otherEnvPathKey = Object.keys(customEnvVars).find((k) => k.toLowerCase() == 'path');
+        if (otherEnvPathKey && customEnvVars[otherEnvPathKey]) {
+            this.envVarsService.appendPath(mergedVars, customEnvVars[otherEnvPathKey]!);
         }
         otherEnvPathKey = Object.keys(kernelEnv).find((k) => k.toLowerCase() == 'path');
         if (otherEnvPathKey && kernelEnv[otherEnvPathKey]) {
             this.envVarsService.appendPath(mergedVars, kernelEnv[otherEnvPathKey]!);
         }
-        if (customEditVars.PYTHONPATH) {
```
Comment (DonJayamanne) [inline src/kernels/raw/launcher/kernelEnvVarsService.node.ts:107]: renamed to make is clear that these are `custom Environment variables`. I am guessing the `Edit` was a typo from my end.

File: src/kernels/jupyter/jupyterKernelService.node.ts [hunk 5]
```diff
@@ -276,75 +274,35 @@ export class JupyterKernelService implements IJupyterKernelService {
                     );
                     specModel.argv[0] = interpreter.uri.fsPath;
                 }
-                // Get the activated environment variables (as a work around for `conda run` and similar).
-                // This ensures the code runs within the context of an activated environment.
-                const interpreterEnv = await this.activationHelper
```
Comment (DonJayamanne) [inline src/kernels/jupyter/jupyterKernelService.node.ts:281]: Removed duplicate code in favor of `kernelEnvVars.getEnvironmentVariables`
Now both raw and jupyter kernels (both) use the exact same code to generate env variables for kernels.

**Review by IanMatthewHuff — APPROVED** [review]

**Comment by jtele2** [comment]
I still cannot get this to work. How do I set environment variables from a `.env` file for use within Python Interactive (Jupyter) using `#%%` cell distinguishers with `.py` files? Does this fix it?

**Pull request merged**